rclcpp/src/rclcpp/parameter_service.cpp

```diff
@@ -41,12 +41,13 @@ ParameterService::ParameterService(
       const std::shared_ptr<rcl_interfaces::srv::GetParameters::Request> request,
       std::shared_ptr<rcl_interfaces::srv::GetParameters::Response> response)
     {
-      for (const auto & name : request->names) {
-        // Default construct param to NOT_SET
-        rclcpp::Parameter param;
-        node_params->get_parameter(name, param);
-        // push back NOT_SET when get_parameter() call fails
-        response->values.push_back(param.get_value_message());
+      try {
+        auto parameters = node_params->get_parameters(request->names);
+        for (const auto param : parameters) {
+          response->values.push_back(param.get_value_message());
+        }
+      } catch (const rclcpp::exceptions::ParameterNotDeclaredException & ex) {
+        RCLCPP_WARN(rclcpp::get_logger("rclcpp"), "Failed to get parameters: %s", ex.what());
```

Comment:
why logging a warning when an external request used the wrong name?
IMO, this should be ignored silently.

Reply:
i think that it would be nice to show warning on the node that's been requested with `non-exist` parameter name when allow_undeclare_parameter is false on that node.

Reply:
A warning is logged if the node is malfunctioning.

In this case, the node received a "bad service request" so in the response it should be encoded that the "request is bad".
But there's nothing malfunctioning, somebody else sent a "bad request", so no need to log a warn.

Reply:
that is correct, it makes sense. i will change these warnings into debug information.